### PR TITLE
Reaxpro work service

### PR DESCRIPTION
**Pull request by Kirankumaraswamy**
Changes for addtional endpoints required by reaxpro-workflow.

### CODE REVIEW

**Comment by pablo-de-andres** [comment]
I thought we discussed using one unique endpoint for uploading data, and a potential parameter to flag wether it is cache or persistent?

**Comment by Kirankumaraswamy** [comment]
I need to discuss with Matthias about this. In his implementation he doesn't have a collection and also he is using uuid as a unique parameter. I have to suggest the name change to dataset_name instead.

File: marketplace_standard_app_api/models/transformation.py [hunk 1]
```diff
@@ -67,3 +72,14 @@ class TransformationStateResponse(BaseModel):
...
+
+class RegisteredModels(BaseModel):
+    message: Optional[str] = None
+    registered_models: List[ModelName]
+
+
+class NewModel(BaseModel):
+    parameters: dict
+
+    modelname: NewTransformationStates = TransformationState.CREATED
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/models/transformation.py:85]: Why do you need a new model? Can yo unot use the `NewTransformationModel`? And why is the `modelname` a state?

File: marketplace_standard_app_api/routers/transformation.py [hunk 2]
```diff
@@ -109,9 +111,7 @@ async def update_transformation(
         404: {"description": "Not found."},
     },
 )
-async def get_transformation_state(
-    transformation_id: TransformationId,
-) -> TransformationStateResponse:
+async def get_transformation_state() -> TransformationStateResponse:
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/routers/transformation.py:114]: How are you supposed to get the state if you do not provide an id for the transformation?

File: marketplace_standard_app_api/routers/transformation.py [hunk 3]
```diff
@@ -127,3 +127,62 @@ async def list_transformation(
...
+    summary="Retrieve an example for a model registered in the app.",
+    responses={
+        404: {"description": "Not found."},
+    },
+)
+async def get_example(
+    modelname: ModelName,
+) -> JSONResponse:
+    """Retrieve an example for a model registered in the app."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/routers/transformation.py:174]: I am not too sure about this capability, but:
- The operation (get) should not be encoded in the path.
- For me `/models/{modelname}/example` seems more logical.
- ´getExample´ is a very generic capability name for the very specific task you have in mind

File: marketplace_standard_app_api/routers/transformation.py [hunk 4]
```diff
@@ -127,3 +127,62 @@ async def list_transformation(
...
+    summary="Retreive schema of a model registered in the app.",
+    responses={
+        404: {"description": "Not found."},
+    },
+)
+async def get_schema(
+    modelname: ModelName,
+) -> JSONResponse:
+    """Retreive schema of a model registered in the app."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/routers/transformation.py:159]: See the one below

File: marketplace_standard_app_api/routers/transformation.py (hunk 2 above)
Comment (Kirankumaraswamy) [reply marketplace_standard_app_api/routers/transformation.py:114]: This is a mistake. I was not intended to remove it. I will add it again.

File: marketplace_standard_app_api/routers/transformation.py (hunk 3 above)
Comment (Kirankumaraswamy) [reply marketplace_standard_app_api/routers/transformation.py:174]: This endpoint is specific to reaxpro-workflow usecase and hence I used the same name provided by Matthias. I will change the definition as you suggested.

File: marketplace_standard_app_api/routers/system.py [hunk 5]
```diff
@@ -36,3 +36,33 @@ async def global_search(
...
+    operation_id="getInfo",
+    summary="Returns system information.",
+    responses={
+        404: {"description": "Not found."},
+        501: {"description": "Not implemented."},
+    },
+)
+async def get_info() -> JSONResponse:
+    """Return the any information related to application. The user can decide the type of information
+    based on the filter parameters (generally query parameters) provided by the application.
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/routers/system.py:52]:
```suggestion
    """Return information related to the application.
    
    The application developer may decide what information, and define their own
    filter parameters (generally query parameters).
```

File: marketplace_standard_app_api/routers/system.py [hunk 6]
```diff
@@ -36,3 +36,33 @@ async def global_search(
...
+    "/logs",
+    operation_id="getLogs",
+    summary="Returns logs from the application.",
+    responses={
+        404: {"description": "Not found."},
+        501: {"description": "Not implemented."},
+    },
+)
+async def get_logs(id: Optional[str], limit: int = 100, offset: int = 0) -> Response:
+    """Return logs from the application."""
```
Comment (pablo-de-andres) [inline marketplace_standard_app_api/routers/system.py:67]:
```suggestion
    """Return application logs.
    
    If an id is provided, the logs will be for a specific entity (transformation, collection or dataset)."""
```

**Comment by pablo-de-andres** [comment]
If you can review those two minor changes, I would merge this to main, and move to review https://github.com/materials-marketplace/python-sdk/pull/70

**Review by pablo-de-andres — APPROVED** [review]

**Pull request merged**